src/frontend/packages/activitypub-components/src/hooks/useCollection.ts

```diff
@@ -104,29 +122,51 @@ const useCollection = (predicateOrUrl: string, options: UseCollectionOptions = {
           setYieldsTotalItems(true);
         } else if (!localIsPaginated) {
           // If the collection is not paginated, we can count items
-          const items = arrayOf(json.orderedItems || json.items);
+          const items = arrayOf(json[itemsKey]);
           if (items) setTotalItems(items.length);
         }
       }
 
-      // If first page, handle this here.
-      if ((json.type === 'OrderedCollection' || json.type === 'Collection') && json.first) {
-        if (json.first?.items) {
-          if (json.first?.items.length === 0 && json.first?.next) {
-            // Special case where the first property is an object without items
-            ({ json } = await dataProvider.fetch(json.first?.next));
-          } else {
-            json = json.first;
+      // Validate the json with the SHACL shape
+      if (shaclShapeUri !== '' && json[itemsKey] && json[itemsKey].length > 0) {
+        try {
+          // get the context from the options or the server's response or fail to validate
+          const context = shaclValidationContext || json['@context'];
+          if (!context) {
+            throw new Error(
+              `No context provided in options.shaclValidationContext and none returned by the server.\nA context is required to expand the collection's items and validate them.`
+            );
```

Comment:
I'm not sure the `shaclValidationContext` option is needed. If the server returns a valid JSON-LD, there will always be a context.